projects/github-actions/required-review/src/requirement.js

```diff
@@ -81,22 +85,42 @@ function buildReviewerFilter( config, teamConfig, indent ) {
 	if ( op === 'any-of' ) {
 		return async function ( reviewers ) {
 			core.info( `${ indent }Union of these:` );
-			return printSet( `${ indent }=>`, [
-				...new Set(
-					( await Promise.all( arg.map( f => f( reviewers, `${ indent }  ` ) ) ) ).flat( 1 )
-				),
-			] );
+			const reviewersAny = await Promise.all( arg.map( f => f( reviewers, `${ indent }  ` ) ) );
+			const requirementsMet = [];
+			const neededTeams = [];
+			for ( const requirementResult of reviewersAny ) {
+				if ( requirementResult.teamReviewers.length != 0 ) {
+					requirementsMet.push( requirementResult.teamReviewers );
+				};
+				if ( requirementResult.neededTeams.length != 0 ) {
+					neededTeams.push( requirementResult.neededTeams );
+				};
+			};
+			if( requirementsMet.some( a => a.length > 0 ) ){ // If there are requirements met, zero out the needed teams
```

Comment:
The way you have the code above written, every element of `requirementsMet` will have `.length > 0`. So as long as `requirementsMet.length > 0`, we're good here.